modules/core/src/main/scala/fs2/kafka/consumer/KafkaConsumeChunk.scala

```diff
@@ -0,0 +1,88 @@
+import fs2.kafka.CommittableConsumerRecord
+import fs2.kafka.CommittableOffsetBatch
+import fs2.kafka.ConsumerRecord
+
+trait KafkaConsumeChunk[F[_], K, V] extends KafkaConsume[F, K, V] {
+
+  /**
+    * Consume from all assigned partitions concurrently, processing the records in `Chunk`s. For
+    * each `Chunk`, the provided `processor` is called, after that has finished the offsets for all
+    * messages in the chunk are committed.<br><br>
+    *
+    * This method is intended to be used in cases that require at-least-once-delivery, where
+    * messages have to be processed before offsets are committed. By relying on the methods like
+    * [[partitionedStream]], [[records]], and similar, you have to correctly implement not only your
+    * processing logic but also the correct mechanism for committing offsets. This can be tricky to
+    * do in a correct and efficient way.<br><br>
+    *
+    * Working with `Chunk`s of records has several benefits:<br>
+    *   - As a user, you don't have to care about committing offsets correctly. You can focus on
+    *     implementing your business logic<br>
+    *   - It's very straightforward to batch several messages from a `Chunk` together, e.g. for
+    *     efficient writes to a persistent storage<br>
+    *   - You can liberally use logic that involves concurrency, filtering, and re-ordering of
+    *     messages without having to worry about incorrect offset commits<br>
+    *
+    * <br>
+    *
+    * The `processor` is a function that takes a `Chunk[ConsumerRecord[K, V]]` and returns a
+    * `F[CommitNow]`. [[CommitNow]] is isomorphic to `Unit`, but helps in transporting the intention
+    * that processing of a `Chunk` is done, offsets should be committed, and no important processing
+    * should be done afterwards.<br><br>
+    *
+    * The returned value has the type `F[Nothing]`, because it's a never-ending process that doesn't
+    * terminate, and therefore doesn't return a result.
+    *
+    * @note
+    *   This method does not make any use of Kafka's auto-commit feature, it implements "manual"
+    *   commits in a way that suits most of the common use cases.
+    * @note
+    *   you have to first use `subscribe` or `assign` the consumer before using this `Stream`. If
+    *   you forgot to subscribe, there will be a [[NotSubscribedException]] raised in the `Stream`.
+    * @see
+    *   [[partitionedStream]]
+    * @see
+    *   [[CommitNow]]
+    */
+  final def consumeChunk(
+    processor: Chunk[ConsumerRecord[K, V]] => F[CommitNow]
+  )(implicit F: Concurrent[F]): F[Nothing] = partitionedStream
+    .map(
+      _.chunks.evalMap(consumeChunk(processor))
+    )
+    .parJoinUnbounded
+    .compile
+    .drain >> F.never
+
+  private def consumeChunk(processor: Chunk[ConsumerRecord[K, V]] => F[CommitNow])(
+    chunk: Chunk[CommittableConsumerRecord[F, K, V]]
+  )(implicit F: Monad[F]): F[Unit] = {
+    val (offsets, records) = chunk.foldLeft(
```

Comment:
Great work! I'm not an expert in fs2, but perhaps this will be more memory efficient than allocating chunks to each element?
```suggestion
    val (offsets, records) = chunk
      .mapAccumulate(CommittableOffsetBatch.empty)((offsetBatch, cRecord) =>
        (offsetBatch.updated(cRecord.offset), cRecord.record)
      )
```

Reply:
Thank you for the suggestion, I wasn't aware of `mapAccumulate`. To be honest, I can't really tell if this would be more memory efficient, but from the signature I guess it should definitely be `mapAccumulate` and not `foldLeft`

Reply:
@o0hgt0o Thanks again for your suggestion, I changed it to `mapAccumulate`, I like that more than the `foldLeft`